controllers/extension-dns-service/charts/internal/dns-service-shoot/templates/crds.yaml

```diff
@@ -0,0 +1,41 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: dnsentries.dns.gardener.cloud
```

Comment:
Let's consider a shoot is also a seed cluster and the `dns-service` is enabled. Then the CRD is installed here, but it's also installed by Gardener in its seed controller. Both might overwrite each other. Also, maybe different versions of the `dns-controller-manager`s are used while one version is working only with another CRD.
Long story short: I don't think it's a good idea to use exactly the same `DNSEntry` resource here. Shouldn't we have a separate one for the `dns-service` that cannot conflict with the regular `DNSEntry` of the dns-controller-manager?